### PR TITLE
[move-examples] Add multiple examples for resource accounts

### PR DESCRIPTION
**Pull request by xbtmatt**
### Description
Merged the publishing code from this module with the [ package manager in PR #8641. ](https://github.com/aptos-labs/aptos-core/pull/8641)

Added three sections to the `examples/resource-accounts` section in the docs.

The tutorial shows examples and sample code that you can compile and publish to test it.

### Test Plan
There is a unit test in `upgradeable_resource_account_package.rs` in the unit tests folder:

`RUST_MIN_STACK=4297152 cargo test --package e2e-move-tests --lib -- tests::upgradeable_resource_account_package::code_upgrading_using_resource_account --exact --nocapture`

The tutorial shows a more complete example of how to do it using the `build-publish-package` command to generate a JSON file with the module bytecode.

### CODE REVIEW

File: developer-docs-site/docs/concepts/resource-accounts.md [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+---
+
+## What is a resource account?
+
+A resource account is an [Account](https://aptos.dev/concepts/accounts/) that's used to store and manage resources. It can be a simple storage account that's used merely to separate different resources for an account or a module, or it can be utilized to programmatically control resource management in a contract.
+
+There are two distinct ways to manage a resource account:
+
+1. The auth key is rotated to a separate account that can control it manually
+2. The Move VM rotates the auth key to 0x0 and controls the account through a SignerCapability
```
Comment (davidiw) [inline developer-docs-site/docs/concepts/resource-accounts.md:13]: what's an auth key?

File: developer-docs-site/docs/concepts/resource-accounts.md [hunk 2]
```diff
@@ -0,0 +1,387 @@
+---
+title: "Resource Accounts"
+id: "resource-accounts"
+---
+
```
Comment (davidiw) [inline developer-docs-site/docs/concepts/resource-accounts.md:5]: I think you need a title `#`

File: developer-docs-site/docs/concepts/resource-accounts.md [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+}
+```
+
+When this function is called, the auth key of the resource account is rotated to `0x0`, which gives the Move VM the capability to generate the resource account's signer from a `SignerCapability`. You can store this `SignerCapability` and retrieve it later to sign for the resource account.
+
+This is often integral to automating a smart contract in Move. It gives the developer the ability to generate a signer for an account programmatically.
+
+Notice that the creation function returns the resource account's `signer` and a `SignerCapability` resource. Let's discuss what a `SignerCapability` is and then how to store it.
+
+#### What's a SignerCapability?
```
Comment (davidiw) [inline developer-docs-site/docs/concepts/resource-accounts.md:64]: headers are weird 2 -> 4 ?

File: developer-docs-site/docs/concepts/resource-accounts.md [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+
+Call the function, but change the sender account to appear as the resource account with the CLI flag `--sender-account`. If the source address is the `default` profile:
+
+```shell
+aptos move run --function-id MODULE_ADDRESS::MODULE_NAME::retrieve_cap --args address:default --sender-account RESOURCE_ADDRESS_HERE --profile default
+```
+
+## How is the address for a resource account derived?
+
+When a resource account is created, the address is derived from a SHA3-256 hash of the requesting account's address plus an optional byte vector `seed`. If you want to know the resource address generated by an account + a given arbitrary seed, you can call the `create_resource_address` function in `account.move`:
```
Comment (davidiw) [inline developer-docs-site/docs/concepts/resource-accounts.md:381]: we should give the actual pseudo code, this is insufficient.

**Review by davidiw — COMMENTED** [review]
Could we merge this pr https://github.com/aptos-labs/aptos-core/pull/8641#pullrequestreview-1509687663 here?

@movekevin

File: aptos-move/e2e-move-tests/src/tests/upgrade_resource_contract.rs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{assert_success, tests::common, MoveHarness};
+use aptos_types::{
+    account_address::{create_resource_address, AccountAddress},
+};
+
+// Note: this module uses parameterized tests via the
+// [`rstest` crate](https://crates.io/crates/rstest)
+// to test for multiple feature combinations.
```
Comment (davidiw) [inline aptos-move/e2e-move-tests/src/tests/upgrade_resource_contract.rs:11]: liar, liar, pants on fire.

**Review by davidiw — CHANGES_REQUESTED** [review]
requesting changes so I don't open this again until it is ready, feel free to dismiss or poke me for a review

**Comment by xbtmatt** [comment]
@saharct Wasn't sure how to structure the subsections, there's a lot of content in there. I also feel some of it could go in a different section, but tried to organize it so it's all together. Tagging you in case you want to check it

File: developer-docs-site/docs/guides/resource-accounts/common-questions.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Common questions
+
+## What is a resource account?
+
+A [resource account](../../move/move-on-aptos/resource-accounts.md) is an [Account](../../concepts/accounts/) that's used to store and manage resources independent of a user. It can be used as a simple storage account or it can be utilized to programmatically manage resources from within a smart contract.
```
Comment (0xmaayan) [inline developer-docs-site/docs/guides/resource-accounts/common-questions.md:13]: can we link somewhere that explains what is a resource?

File: developer-docs-site/docs/guides/resource-accounts/common-questions.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+## What is a resource account?
+
+A [resource account](../../move/move-on-aptos/resource-accounts.md) is an [Account](../../concepts/accounts/) that's used to store and manage resources independent of a user. It can be used as a simple storage account or it can be utilized to programmatically manage resources from within a smart contract. 
+
+## How are resource accounts created?
+
+Let's review the two functions used to create resource accounts and what they return.
+
+First off, note that both creation functions allow for the input of a `seed` byte vector for the [ensuing hash used to compute the resource address](#how-is-the-address-for-a-resource-account-derived).
```
Comment (0xmaayan) [inline developer-docs-site/docs/guides/resource-accounts/common-questions.md:19]: maybe a brief explanation on what is a seed?

File: developer-docs-site/docs/guides/resource-accounts/common-questions.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
+---
+title: "Common questions"
+id: "common-questions"
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Common questions
```
Comment (0xmaayan) [inline developer-docs-site/docs/guides/resource-accounts/common-questions.md:9]: maybe add some use cases on why we want to create and use a resource account?

File: developer-docs-site/docs/guides/resource-accounts/common-questions.md (hunk 7 above)
Comment (xbtmatt) [reply developer-docs-site/docs/guides/resource-accounts/common-questions.md]: Done!

File: developer-docs-site/docs/guides/resource-accounts/common-questions.md (hunk 8 above)
Comment (xbtmatt) [reply developer-docs-site/docs/guides/resource-accounts/common-questions.md:9]: Good point, maybe went overboard with it but hey, it's useful info!

**Pull request closed without merging**